From fc0a2ff5fdc888118d2d8270adac04fc71e1f12a Mon Sep 17 00:00:00 2001
From: andregoetz <andregoetz37@gmail.com>
Date: Fri, 18 Feb 2022 19:51:43 +0100
Subject: [PATCH] add cycle_keep_windows_key and select_key

---
 libxfce4kbd-private/xfce-shortcuts-xfwm4.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libxfce4kbd-private/xfce-shortcuts-xfwm4.c b/libxfce4kbd-private/xfce-shortcuts-xfwm4.c
index 9caf627..d1f19a4 100644
--- a/libxfce4kbd-private/xfce-shortcuts-xfwm4.c
+++ b/libxfce4kbd-private/xfce-shortcuts-xfwm4.c
@@ -39,7 +39,9 @@ const ShortcutTemplate xfwm4_shortcut_values[] = {
   { N_("Left"), "left_key" },
   { N_("Right"), "right_key" },
   { N_("Cancel"), "cancel_key" },
+  { N_("Select"), "select_key" },
   { N_("Cycle windows"), "cycle_windows_key" },
+  { N_("Cycle windows (Keep open)"), "cycle_keep_windows_key" },
   { N_("Cycle windows (Reverse)"), "cycle_reverse_windows_key" },
   { N_("Switch window for same application"), "switch_window_key" },
   { N_("Switch application"), "switch_application_key" },
-- 
2.35.1

